decidim-core/lib/tasks/decidim_webpacker_tasks.rake

```diff
@@ -73,32 +69,57 @@ namespace :decidim do
     end
 
     def decidim_npm_packages
+      gem_path = unreleased_gem_path
+
+      if gem_path
+        package_spec = "./packages/%s"
+
+        # The packages folder needs to be copied to the application folder
+        # because the linked dependencies are not installed when packages
+        # are installed using file references outside the application root
+        # where the `package.json` is located at. For more information, see:
+        # https://github.com/npm/cli/issues/2339
+        FileUtils.rm_rf(rails_app_path.join("packages"))
+        FileUtils.cp_r(gem_path.join("packages"), rails_app_path)
+      else
+        package_spec = "@decidim/%s@~#{Decidim::GemManager.semver_friendly_version(decidim_gemspec.version.to_s)}"
+      end
+
+      local_npm_dependencies.transform_values { |names| names.map { |name| format(package_spec, name) } }
+    end
+
+    def unreleased_gem_path
       if decidim_gemspec.source.is_a?(Bundler::Source::Rubygems)
-        if released_version?
-          return {
-            dev: "@decidim/dev@~#{Decidim::GemManager.semver_friendly_version(decidim_gemspec.version.to_s)}",
-            prod: "@decidim/all@~#{Decidim::GemManager.semver_friendly_version(decidim_gemspec.version.to_s)}"
-          }
-        else
-          gem_path = Pathname(decidim_gemspec.full_gem_path)
-        end
+        return if released_version?
+
+        gem_path = Pathname(decidim_gemspec.full_gem_path)
       else
         gem_path = decidim_gemspec.source.path
         gem_path = Pathname(ENV["BUNDLE_GEMFILE"]).dirname.join(gem_path) if gem_path.relative?
       end
 
-      # The packages folder needs to be copied to the application folder
-      # because the linked dependencies are not installed when packages
-      # are installed using file references outside the application root
-      # where the `package.json` is located at. For more information, see:
-      # https://github.com/npm/cli/issues/2339
-      FileUtils.rm_rf(rails_app_path.join("packages"))
-      FileUtils.cp_r(gem_path.join("packages"), rails_app_path)
-
-      {
-        dev: "./packages/dev",
-        prod: "./packages/all"
-      }
+      gem_path
+    end
+
+    def local_npm_dependencies
+      @local_npm_dependencies ||= begin
+        package_json = JSON.parse(File.read(decidim_path.join("package.json")))
+
+        {
+          prod: local_npm_dependencies_list(package_json["dependencies"]),
+          dev: local_npm_dependencies_list(package_json["devDependencies"]),
+          peer: local_npm_dependencies_list(package_json["peerDependencies"]),
+          optional: local_npm_dependencies_list(package_json["optionalDependencies"])
```

Comment:
@leio10 If you do it this way, you can skip the `:peer` and `:optional` categories here.

Peer dependencies are used for NPM dependencies such as eslint/stylelint configs to mark the version of the host package they are compatible with.

Optional dependencies could be used e.g. in a situation where you add additional non-necessary functionality to a package, e.g. a package that exports data to CSV, you could add also XLSX format as optional dependency (and functionality available only when the optional dependency is met). Currently there are no optional dependencies within the Decidim's own NPM packages.

Decidim's root `package.json` does not define either of these and never will. These categories are meant for NPM packages that are pushed to a repository.

And they should not be added to the application's `package.json` either.

So, these two lines can be removed.